Reconstruct HTTP Result in safe way using Result.copy().

helllamer opened the pull request:
This prevents losing some parts of Result like session and other.
Fixes #64

jleider commented on app/com/mohiva/play/compressor/CompressorFilter.scala:
```diff
@@ -88,17 +88,31 @@ abstract class CompressorFilter[C <: Compressor] extends Filter {
-              result.header.copy(headers = result.header.headers),
-              HttpEntity.Streamed(Source.single(ByteString(compressed)), Some(length.toLong), result.body.contentType)
+            result.copy(
+              //header = result.header.copy(headers = result.header.headers),
```
Did you mean to leave this commented out?

helllamer replied:
Hmm, this code previously was UNcommented. But i think, this code makes no sense.
May be, something was or invalid missing here?

helllamer commented:
Ok, i've removed couple of useless lines from project. Not so much to review.

akkie commented on app/com/mohiva/play/compressor/CompressorFilter.scala:
```diff
@@ -88,17 +88,30 @@ abstract class CompressorFilter[C <: Compressor] extends Filter {
-          Future.successful(Result(result.header, HttpEntity.Strict(ByteString(compress(data)), contentType)))
-        case HttpEntity.Streamed(data, _, contentType) =>
-          data.toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run() map { bytes =>
+
```
I've stumbled over the same issue in Silhouette. Could you please change the code to something like this:
```scala
      result.body match {
        case body: HttpEntity.Strict =>
          Future.successful(result.copy(
            body = body.copy(
              data = ByteString(compress(body.data))
            )
          ))
        case body: HttpEntity.Streamed =>
          for {
            bytes <- body.data.toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run()
          } yield {
            val compressed = compress(bytes)
            val length = compressed.length
            result.copy(
              body = body.copy(
                data = Source.single(ByteString(compressed)),
                contentLength = Some(length.toLong)
              )
            )
          }
        case _ => 
          Future.successful(result)
      }
```

Appending number to variables is a code smell in my eyes :wink:

helllamer commented:
Ok, the *last* fix.

akkie commented on app/com/mohiva/play/compressor/CompressorFilter.scala:
```diff
@@ -88,15 +88,25 @@ abstract class CompressorFilter[C <: Compressor] extends Filter {
-          Future.successful(Result(result.header, HttpEntity.Strict(ByteString(compress(data)), contentType)))
-        case HttpEntity.Streamed(data, _, contentType) =>
-          data.toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run() map { bytes =>
+        case body0: HttpEntity.Strict =>
```
Why do you use `body0` instead of `body`?

helllamer commented:
Why you are spending minutes your (and my) time for this endless comments and bla-bla instead of just-fix-it-as-YOU-want in several seconds?

akkie commented:
First of all thanks for your contribution. The reason is that I try to keep the code quality high. I've explained it yesterday, that in my opinion,  attaching numbers to variable names is a code smell. So I don't understand why you attach the numbers again?

helllamer commented:
1. Frankly speaking, I don't see a "code quality" in the sources. Maybe, we a read different sources?
Some code lines looking like "minified js", where mostly all newlines and spaces are stripped out with fences of braces.
During years, this silently causes crazy errors like #65 and deadly useless code like `result.header.copy(headers = result.header.headers)`, because very-long-horizontal-lines-what-do-everything are hard to read by 3rd-party people, most people prefer not to read such code, only to patch it.

2. Second problem, i already implemented what i needed, and modified the code by your, guys, non-constructive wishes. But, let's stop here. After second code-style-wish, it was clear to me: code is already ok, problem not in the code.

akkie commented:
Thanks for your contribution. I'll change the lines by myself.
